File: Python/350. Intersection of Two Arrays II.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        fq=Counter(nums1)
        res=[]

        for n in nums2:
            if fq[n]>0:
                res.append(n)
                fq[n]-=1
        return res
        

File: Java/350. Intersection of Two Arrays II.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
     HashMap<Integer,Integer> map=new HashMap<>();
     ArrayList<Integer> l=new ArrayList<>();

     for(int n:nums1)
     {
        map.put(n,map.getOrDefault(n,0)+1);
     }
     for(int n:nums2)
     {
        if(map.containsKey(n)&&map.get(n)>0)
        {
            l.add(n);
            map.put(n,map.getOrDefault(n,0)-1);
        }
     }
     int i=0;
          int[] re=new int[l.size()];
          for(int n:l)
          {
            re[i++]=n;
          }
          return re;
    }
}